### PR TITLE
Register to challenge

### PR DESCRIPTION
**Pull request by victorbalan**

### CODE REVIEW

File: platform-server-kernel/src/main/java/uno/cod/platform/server/core/repository/ChallengeRepository.java [hunk 1]
```diff
@@ -34,15 +35,25 @@ Challenge findOneByTemplateAndStartDateAndOrganization(@Param("templateId") UUID
             "WHERE users.id = :user")
     List<Challenge> findAllByInvitedUser(@Param("user") UUID id);
 
+    // FUCK HIBERNATE users.id <> user , users.id!=: user , users.id not in (:user) does not work
```
Comment (flowlo) [inline platform-server-kernel/src/main/java/uno/cod/platform/server/core/repository/ChallengeRepository.java]: Really?

File: platform-server-kernel/src/main/java/uno/cod/platform/server/core/repository/ChallengeRepository.java [hunk 2]
```diff
@@ -37,12 +38,20 @@ Challenge findOneByTemplateAndStartDateAndOrganization(@Param("templateId") UUID
     @Query("SELECT challenge FROM Challenge challenge " +
             "LEFT JOIN FETCH challenge.challengeTemplate challengeTemplate " +
             "LEFT JOIN FETCH challengeTemplate.organization organization " +
-            "WHERE challenge.inviteOnly= :inviteOnly AND challenge.endDate > :endDate")
-    List<Challenge> findAllByInviteOnlyAndEndDateAfter(@Param("inviteOnly") Boolean inviteOnly, @Param("endDate")ZonedDateTime endDate);
+            "WHERE challenge.inviteOnly = :inviteOnly AND challenge.endDate > :endDate " +
+            "AND challenge.id NOT IN(" +
+            "SELECT challenge2.id FROM Challenge challenge2 " +
```
Comment (Exordian) [inline platform-server-kernel/src/main/java/uno/cod/platform/server/core/repository/ChallengeRepository.java]: please avoid subqueries when possible, those can be very expensive

File: platform-server-kernel/src/main/resources/i18n/exceptions_en_US.properties [hunk 3]
```diff
@@ -38,4 +38,7 @@ team.has.member = User is already in the team
 team.user.already.invited = User is already invited
 team.invite.notfound = No invite for this team found
 
-token.invalid=Invalid Token
\ No newline at end of file
+token.invalid=Invalid Token
+
+already.registered.to.challenge = Already registered to challenge
```
Comment (Exordian) [inline platform-server-kernel/src/main/resources/i18n/exceptions_en_US.properties]: please start with the name of the module i.e. "challenge"

File: platform-server-kernel/src/main/java/uno/cod/platform/server/core/repository/ChallengeRepository.java (hunk 2 above)
Comment (victorbalan) [inline platform-server-kernel/src/main/java/uno/cod/platform/server/core/repository/ChallengeRepository.java]: it`s a hibernate fault i cannot join and search for `userId != user.id` or  `userId <> user.id` or `userId NOT IN (user.id)
`

Comment (victorbalan) [inline platform-server-kernel/src/main/java/uno/cod/platform/server/core/repository/ChallengeRepository.java]: https://hibernate.atlassian.net/browse/HHH-3022

**Pull request closed without merging**